// react-day-picker/dist/cjs/locale.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.hu = exports.ht = exports.hr = exports.hi = exports.he = exports.gu = exports.gl = exports.gd = exports.fy = exports.frCH = exports.frCA = exports.fr = exports.fi = exports.faIRJalali = exports.faIR = exports.eu = exports.et = exports.es = exports.eo = exports.enZA = exports.enUSJalali = exports.enUS = exports.enNZ = exports.enIN = exports.enIE = exports.enGB = exports.enCA = exports.enAU = exports.el = exports.deAT = exports.de = exports.da = exports.cy = exports.cs = exports.ckb = exports.ca = exports.bs = exports.bn = exports.bg = exports.beTarask = exports.be = exports.az = exports.arTN = exports.arSA = exports.arMA = exports.arEG = exports.arDZ = exports.ar = exports.amET = exports.af = void 0;
exports.zhTW = exports.zhHK = exports.zhCN = exports.vi = exports.uzCyrl = exports.uz = exports.uk = exports.ug = exports.tr = exports.th = exports.te = exports.ta = exports.sv = exports.srLatn = exports.sr = exports.sq = exports.sl = exports.sk = exports.se = exports.ru = exports.ro = exports.ptBR = exports.pt = exports.pl = exports.oc = exports.nn = exports.nlBE = exports.nl = exports.nb = exports.mt = exports.ms = exports.mn = exports.mk = exports.lv = exports.lt = exports.lb = exports.ko = exports.kn = exports.km = exports.kk = exports.ka = exports.jaHira = exports.ja = exports.itCH = exports.it = exports.is = exports.id = exports.hy = void 0;
__exportStar(require("date-fns/locale"), exports);
/**
 * DayPicker locales extended with translated labels.
 *
 * @since V9.12.0
 */
var af_js_1 = require("./locale/af.js");
Object.defineProperty(exports, "af", { enumerable: true, get: function () { return af_js_1.af; } });
var am_ET_js_1 = require("./locale/am-ET.js");
Object.defineProperty(exports, "amET", { enumerable: true, get: function () { return am_ET_js_1.amET; } });
var ar_js_1 = require("./locale/ar.js");
Object.defineProperty(exports, "ar", { enumerable: true, get: function () { return ar_js_1.ar; } });
var ar_DZ_js_1 = require("./locale/ar-DZ.js");
Object.defineProperty(exports, "arDZ", { enumerable: true, get: function () { return ar_DZ_js_1.arDZ; } });
var ar_EG_js_1 = require("./locale/ar-EG.js");
Object.defineProperty(exports, "arEG", { enumerable: true, get: function () { return ar_EG_js_1.arEG; } });
var ar_MA_js_1 = require("./locale/ar-MA.js");
Object.defineProperty(exports, "arMA", { enumerable: true, get: function () { return ar_MA_js_1.arMA; } });
var ar_SA_js_1 = require("./locale/ar-SA.js");
Object.defineProperty(exports, "arSA", { enumerable: true, get: function () { return ar_SA_js_1.arSA; } });
var ar_TN_js_1 = require("./locale/ar-TN.js");
Object.defineProperty(exports, "arTN", { enumerable: true, get: function () { return ar_TN_js_1.arTN; } });
var az_js_1 = require("./locale/az.js");
Object.defineProperty(exports, "az", { enumerable: true, get: function () { return az_js_1.az; } });
var be_js_1 = require("./locale/be.js");
Object.defineProperty(exports, "be", { enumerable: true, get: function () { return be_js_1.be; } });
var be_tarask_js_1 = require("./locale/be-tarask.js");
Object.defineProperty(exports, "beTarask", { enumerable: true, get: function () { return be_tarask_js_1.beTarask; } });
var bg_js_1 = require("./locale/bg.js");
Object.defineProperty(exports, "bg", { enumerable: true, get: function () { return bg_js_1.bg; } });
var bn_js_1 = require("./locale/bn.js");
Object.defineProperty(exports, "bn", { enumerable: true, get: function () { return bn_js_1.bn; } });
var bs_js_1 = require("./locale/bs.js");
Object.defineProperty(exports, "bs", { enumerable: true, get: function () { return bs_js_1.bs; } });
var ca_js_1 = require("./locale/ca.js");
Object.defineProperty(exports, "ca", { enumerable: true, get: function () { return ca_js_1.ca; } });
var ckb_js_1 = require("./locale/ckb.js");
Object.defineProperty(exports, "ckb", { enumerable: true, get: function () { return ckb_js_1.ckb; } });
var cs_js_1 = require("./locale/cs.js");
Object.defineProperty(exports, "cs", { enumerable: true, get: function () { return cs_js_1.cs; } });
var cy_js_1 = require("./locale/cy.js");
Object.defineProperty(exports, "cy", { enumerable: true, get: function () { return cy_js_1.cy; } });
var da_js_1 = require("./locale/da.js");
Object.defineProperty(exports, "da", { enumerable: true, get: function () { return da_js_1.da; } });
var de_js_1 = require("./locale/de.js");
Object.defineProperty(exports, "de", { enumerable: true, get: function () { return de_js_1.de; } });
var de_AT_js_1 = require("./locale/de-AT.js");
Object.defineProperty(exports, "deAT", { enumerable: true, get: function () { return de_AT_js_1.deAT; } });
var el_js_1 = require("./locale/el.js");
Object.defineProperty(exports, "el", { enumerable: true, get: function () { return el_js_1.el; } });
var en_AU_js_1 = require("./locale/en-AU.js");
Object.defineProperty(exports, "enAU", { enumerable: true, get: function () { return en_AU_js_1.enAU; } });
var en_CA_js_1 = require("./locale/en-CA.js");
Object.defineProperty(exports, "enCA", { enumerable: true, get: function () { return en_CA_js_1.enCA; } });
var en_GB_js_1 = require("./locale/en-GB.js");
Object.defineProperty(exports, "enGB", { enumerable: true, get: function () { return en_GB_js_1.enGB; } });
var en_IE_js_1 = require("./locale/en-IE.js");
Object.defineProperty(exports, "enIE", { enumerable: true, get: function () { return en_IE_js_1.enIE; } });
var en_IN_js_1 = require("./locale/en-IN.js");
Object.defineProperty(exports, "enIN", { enumerable: true, get: function () { return en_IN_js_1.enIN; } });
var en_NZ_js_1 = require("./locale/en-NZ.js");
Object.defineProperty(exports, "enNZ", { enumerable: true, get: function () { return en_NZ_js_1.enNZ; } });
var en_US_js_1 = require("./locale/en-US.js");
Object.defineProperty(exports, "enUS", { enumerable: true, get: function () { return en_US_js_1.enUS; } });
var en_US_jalali_js_1 = require("./locale/en-US-jalali.js");
Object.defineProperty(exports, "enUSJalali", { enumerable: true, get: function () { return en_US_jalali_js_1.enUSJalali; } });
var en_ZA_js_1 = require("./locale/en-ZA.js");
Object.defineProperty(exports, "enZA", { enumerable: true, get: function () { return en_ZA_js_1.enZA; } });
var eo_js_1 = require("./locale/eo.js");
Object.defineProperty(exports, "eo", { enumerable: true, get: function () { return eo_js_1.eo; } });
var es_js_1 = require("./locale/es.js");
Object.defineProperty(exports, "es", { enumerable: true, get: function () { return es_js_1.es; } });
var et_js_1 = require("./locale/et.js");
Object.defineProperty(exports, "et", { enumerable: true, get: function () { return et_js_1.et; } });
var eu_js_1 = require("./locale/eu.js");
Object.defineProperty(exports, "eu", { enumerable: true, get: function () { return eu_js_1.eu; } });
var fa_IR_js_1 = require("./locale/fa-IR.js");
Object.defineProperty(exports, "faIR", { enumerable: true, get: function () { return fa_IR_js_1.faIR; } });
var fa_IR_jalali_js_1 = require("./locale/fa-IR-jalali.js");
Object.defineProperty(exports, "faIRJalali", { enumerable: true, get: function () { return fa_IR_jalali_js_1.faIRJalali; } });
var fi_js_1 = require("./locale/fi.js");
Object.defineProperty(exports, "fi", { enumerable: true, get: function () { return fi_js_1.fi; } });
var fr_js_1 = require("./locale/fr.js");
Object.defineProperty(exports, "fr", { enumerable: true, get: function () { return fr_js_1.fr; } });
var fr_CA_js_1 = require("./locale/fr-CA.js");
Object.defineProperty(exports, "frCA", { enumerable: true, get: function () { return fr_CA_js_1.frCA; } });
var fr_CH_js_1 = require("./locale/fr-CH.js");
Object.defineProperty(exports, "frCH", { enumerable: true, get: function () { return fr_CH_js_1.frCH; } });
var fy_js_1 = require("./locale/fy.js");
Object.defineProperty(exports, "fy", { enumerable: true, get: function () { return fy_js_1.fy; } });
var gd_js_1 = require("./locale/gd.js");
Object.defineProperty(exports, "gd", { enumerable: true, get: function () { return gd_js_1.gd; } });
var gl_js_1 = require("./locale/gl.js");
Object.defineProperty(exports, "gl", { enumerable: true, get: function () { return gl_js_1.gl; } });
var gu_js_1 = require("./locale/gu.js");
Object.defineProperty(exports, "gu", { enumerable: true, get: function () { return gu_js_1.gu; } });
var he_js_1 = require("./locale/he.js");
Object.defineProperty(exports, "he", { enumerable: true, get: function () { return he_js_1.he; } });
var hi_js_1 = require("./locale/hi.js");
Object.defineProperty(exports, "hi", { enumerable: true, get: function () { return hi_js_1.hi; } });
var hr_js_1 = require("./locale/hr.js");
Object.defineProperty(exports, "hr", { enumerable: true, get: function () { return hr_js_1.hr; } });
var ht_js_1 = require("./locale/ht.js");
Object.defineProperty(exports, "ht", { enumerable: true, get: function () { return ht_js_1.ht; } });
var hu_js_1 = require("./locale/hu.js");
Object.defineProperty(exports, "hu", { enumerable: true, get: function () { return hu_js_1.hu; } });
var hy_js_1 = require("./locale/hy.js");
Object.defineProperty(exports, "hy", { enumerable: true, get: function () { return hy_js_1.hy; } });
var id_js_1 = require("./locale/id.js");
Object.defineProperty(exports, "id", { enumerable: true, get: function () { return id_js_1.id; } });
var is_js_1 = require("./locale/is.js");
Object.defineProperty(exports, "is", { enumerable: true, get: function () { return is_js_1.is; } });
var it_js_1 = require("./locale/it.js");
Object.defineProperty(exports, "it", { enumerable: true, get: function () { return it_js_1.it; } });
var it_CH_js_1 = require("./locale/it-CH.js");
Object.defineProperty(exports, "itCH", { enumerable: true, get: function () { return it_CH_js_1.itCH; } });
var ja_js_1 = require("./locale/ja.js");
Object.defineProperty(exports, "ja", { enumerable: true, get: function () { return ja_js_1.ja; } });
var ja_Hira_js_1 = require("./locale/ja-Hira.js");
Object.defineProperty(exports, "jaHira", { enumerable: true, get: function () { return ja_Hira_js_1.jaHira; } });
var ka_js_1 = require("./locale/ka.js");
Object.defineProperty(exports, "ka", { enumerable: true, get: function () { return ka_js_1.ka; } });
var kk_js_1 = require("./locale/kk.js");
Object.defineProperty(exports, "kk", { enumerable: true, get: function () { return kk_js_1.kk; } });
var km_js_1 = require("./locale/km.js");
Object.defineProperty(exports, "km", { enumerable: true, get: function () { return km_js_1.km; } });
var kn_js_1 = require("./locale/kn.js");
Object.defineProperty(exports, "kn", { enumerable: true, get: function () { return kn_js_1.kn; } });
var ko_js_1 = require("./locale/ko.js");
Object.defineProperty(exports, "ko", { enumerable: true, get: function () { return ko_js_1.ko; } });
var lb_js_1 = require("./locale/lb.js");
Object.defineProperty(exports, "lb", { enumerable: true, get: function () { return lb_js_1.lb; } });
var lt_js_1 = require("./locale/lt.js");
Object.defineProperty(exports, "lt", { enumerable: true, get: function () { return lt_js_1.lt; } });
var lv_js_1 = require("./locale/lv.js");
Object.defineProperty(exports, "lv", { enumerable: true, get: function () { return lv_js_1.lv; } });
var mk_js_1 = require("./locale/mk.js");
Object.defineProperty(exports, "mk", { enumerable: true, get: function () { return mk_js_1.mk; } });
var mn_js_1 = require("./locale/mn.js");
Object.defineProperty(exports, "mn", { enumerable: true, get: function () { return mn_js_1.mn; } });
var ms_js_1 = require("./locale/ms.js");
Object.defineProperty(exports, "ms", { enumerable: true, get: function () { return ms_js_1.ms; } });
var mt_js_1 = require("./locale/mt.js");
Object.defineProperty(exports, "mt", { enumerable: true, get: function () { return mt_js_1.mt; } });
var nb_js_1 = require("./locale/nb.js");
Object.defineProperty(exports, "nb", { enumerable: true, get: function () { return nb_js_1.nb; } });
var nl_js_1 = require("./locale/nl.js");
Object.defineProperty(exports, "nl", { enumerable: true, get: function () { return nl_js_1.nl; } });
var nl_BE_js_1 = require("./locale/nl-BE.js");
Object.defineProperty(exports, "nlBE", { enumerable: true, get: function () { return nl_BE_js_1.nlBE; } });
var nn_js_1 = require("./locale/nn.js");
Object.defineProperty(exports, "nn", { enumerable: true, get: function () { return nn_js_1.nn; } });
var oc_js_1 = require("./locale/oc.js");
Object.defineProperty(exports, "oc", { enumerable: true, get: function () { return oc_js_1.oc; } });
var pl_js_1 = require("./locale/pl.js");
Object.defineProperty(exports, "pl", { enumerable: true, get: function () { return pl_js_1.pl; } });
var pt_js_1 = require("./locale/pt.js");
Object.defineProperty(exports, "pt", { enumerable: true, get: function () { return pt_js_1.pt; } });
var pt_BR_js_1 = require("./locale/pt-BR.js");
Object.defineProperty(exports, "ptBR", { enumerable: true, get: function () { return pt_BR_js_1.ptBR; } });
var ro_js_1 = require("./locale/ro.js");
Object.defineProperty(exports, "ro", { enumerable: true, get: function () { return ro_js_1.ro; } });
var ru_js_1 = require("./locale/ru.js");
Object.defineProperty(exports, "ru", { enumerable: true, get: function () { return ru_js_1.ru; } });
var se_js_1 = require("./locale/se.js");
Object.defineProperty(exports, "se", { enumerable: true, get: function () { return se_js_1.se; } });
var sk_js_1 = require("./locale/sk.js");
Object.defineProperty(exports, "sk", { enumerable: true, get: function () { return sk_js_1.sk; } });
var sl_js_1 = require("./locale/sl.js");
Object.defineProperty(exports, "sl", { enumerable: true, get: function () { return sl_js_1.sl; } });
var sq_js_1 = require("./locale/sq.js");
Object.defineProperty(exports, "sq", { enumerable: true, get: function () { return sq_js_1.sq; } });
var sr_js_1 = require("./locale/sr.js");
Object.defineProperty(exports, "sr", { enumerable: true, get: function () { return sr_js_1.sr; } });
var sr_Latn_js_1 = require("./locale/sr-Latn.js");
Object.defineProperty(exports, "srLatn", { enumerable: true, get: function () { return sr_Latn_js_1.srLatn; } });
var sv_js_1 = require("./locale/sv.js");
Object.defineProperty(exports, "sv", { enumerable: true, get: function () { return sv_js_1.sv; } });
var ta_js_1 = require("./locale/ta.js");
Object.defineProperty(exports, "ta", { enumerable: true, get: function () { return ta_js_1.ta; } });
var te_js_1 = require("./locale/te.js");
Object.defineProperty(exports, "te", { enumerable: true, get: function () { return te_js_1.te; } });
var th_js_1 = require("./locale/th.js");
Object.defineProperty(exports, "th", { enumerable: true, get: function () { return th_js_1.th; } });
var tr_js_1 = require("./locale/tr.js");
Object.defineProperty(exports, "tr", { enumerable: true, get: function () { return tr_js_1.tr; } });
var ug_js_1 = require("./locale/ug.js");
Object.defineProperty(exports, "ug", { enumerable: true, get: function () { return ug_js_1.ug; } });
var uk_js_1 = require("./locale/uk.js");
Object.defineProperty(exports, "uk", { enumerable: true, get: function () { return uk_js_1.uk; } });
var uz_js_1 = require("./locale/uz.js");
Object.defineProperty(exports, "uz", { enumerable: true, get: function () { return uz_js_1.uz; } });
var uz_Cyrl_js_1 = require("./locale/uz-Cyrl.js");
Object.defineProperty(exports, "uzCyrl", { enumerable: true, get: function () { return uz_Cyrl_js_1.uzCyrl; } });
var vi_js_1 = require("./locale/vi.js");
Object.defineProperty(exports, "vi", { enumerable: true, get: function () { return vi_js_1.vi; } });
var zh_CN_js_1 = require("./locale/zh-CN.js");
Object.defineProperty(exports, "zhCN", { enumerable: true, get: function () { return zh_CN_js_1.zhCN; } });
var zh_HK_js_1 = require("./locale/zh-HK.js");
Object.defineProperty(exports, "zhHK", { enumerable: true, get: function () { return zh_HK_js_1.zhHK; } });
var zh_TW_js_1 = require("./locale/zh-TW.js");
Object.defineProperty(exports, "zhTW", { enumerable: true, get: function () { return zh_TW_js_1.zhTW; } });
